core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -279,12 +285,37 @@ oppia.factory('UnitsObjectFactory', [function() {
     return unit.trim();
   };
 
+  Units.createCurrencyUnits = function() {
+    // Creates user-defined currency (base + sub) units.
+    math.createUnit('dollar', {aliases: [
+      'dollars', 'Dollar', 'USD', '$', 'Dollars']});
+    math.createUnit('cent', {definition: '0.01 dollar', aliases: [
+      'Cent', 'cents', 'Cents']});
+    math.createUnit('rupees', {aliases: [
+      'Rupees', 'Rs', 'rupee', 'Rupee', '₹']});
+    math.createUnit('paise', {definition: '0.01 rupees', aliases: ['paisa']});
+  };
+
   Units.fromRawInputString = function(units) {
     units = units.replace(/per/g, '/');
-    // Right now, validation of currency units is not possible as we need to add
-    // them first.
-    if (units !== '' && !units.includes('$') && !units.includes('Rs') &&
-      !units.includes('₹')) {
+    try {
+      Units.createCurrencyUnits();
+    } catch (parsingError) {}
+
+    // Special symbols need to be replaced as math.js doesn't support custom
+    // units starting with special symbols. Also, it doesn't allow units
+    // followed by a number as in the case of currency units.
+    if (units.includes('$')) {
+      units = units.replace('$', '');
+      units = 'dollar ' + units;
+    }
+    if (units.includes('Rs') || units.includes('₹')) {
```

Comment:
Ditto.

Reply:
Is `Rs` not supported by math.js? I am asking because it is not a special symbol.

Reply:
Yes, it is supported. But `math.js` doesn't allow any unit followed by value. Like: `Rs 5` isn't allowed. So I've to change it as well.